lib/streamlit/util.py

```diff
@@ -403,22 +404,24 @@ def is_function(x):
 
 
 def get_streamlit_file_path(*filepath):
-    """Return the full path to a filepath in ~/.streamlit.
+    """Return the full path to a file in ~/.streamlit.
 
-    Creates ~/.streamlit if needed.
+    This doesn't guarantee that the file (or its directory) exists.
     """
     # os.path.expanduser works on OSX, Linux and Windows
     home = os.path.expanduser("~")
     if home is None:
         raise RuntimeError("No home directory.")
 
-    folder_path = filepath[:-1]
-    st_path = os.path.join(home, STREAMLIT_ROOT_DIRECTORY, *folder_path)
+    return os.path.join(home, STREAMLIT_ROOT_DIRECTORY, *filepath)
```

Comment:
This is not your doing, but rename STREAMLIT_ROOT_DIRECTORY to CONFIG_DIRECTORY_NAME (or FOLDER) since: (1) the "ROOT" part is confusing; (2) it's unclear that this is just a dir name, not a fullpath; (3) we already know that this is Streamlit-related -- no need to say "STREAMLIT"